README.rst

```diff
@@ -385,6 +385,34 @@ The cached values are updated automatically to reflect actual data from database
 the duration of caching these objects using
 `OPENWISP_NOTIFICATIONS_CACHE_TIMEOUT setting <#OPENWISP_NOTIFICATIONS_CACHE_TIMEOUT>`_.
 
+Cache invalidation
+~~~~~~~~~~~~~~~~~~
+
+The function ``register_notification_cache_update`` can be used to register a signal of a model which is being used as an
+``actor``, ``action_object`` and ``target`` objects. As these values are cached for the optimization purpose so their cached 
+values are need to be changed when they are changed. You can register any signal you want which will delete the cached value.
+To register a signal you need to include following code in your ``apps.py``.
+
+.. code-block:: python
+
+    from django.db.models.signals import post_save
+    from swapper import load_model
+
+    def ready(self):
+        super().ready()
+
+        # Include lines after this inside
+        # ready function of you app config class
+        from openwisp_notifications.handlers import register_notification_cache_update
+
+        model = load_model('app_name', 'model_name')
+        register_notification_cache_update(model, post_save, signal_uid="myapp_mymodel_notification_cache_invalidation")
+
+**Note**: You need to import ``register_notification_cache_update`` inside the ``ready`` function or 
+you can define another funtion to register signals which will be called in ``ready`` and then it will be 
+imported in this function. Also ``signal_uid`` is unique identifier of a signal. You can pass any 
+value you want but it is suggested to pass signal name as a string. By default it is ``None``.
```

Comment:
> I think we should add one more test that uses a custom signal. This will check the purpose this feature was added for.

@pandafy Should I use custom signal instead of ``post_save`` or I have to do something else?

Reply:
Use a custom signal for a test in sample app.